src/api/pool.go

```diff
@@ -0,0 +1,109 @@
+package api
+
+import (
+	"context"
+	"github.com/go-resty/resty/v2"
+	"golang.org/x/sync/errgroup"
+	"runtime"
+	"sync"
+)
+
+type Pool struct {
```

Comment:
V tomto projekte potrebujeme hromadne stahovat data zo storage API.

Nacita sa napr. `branch` v nej sa najdu vsetky `configs` v kazdom configu `configRows` ... 
Aby to islo jednoducho a rychlo, tak som pripravil tuto sktrukturu `pool`.

Zakladom je funkcia `processor`, ktora umoznuje lubovolne spracovat `response` a pripadne poslat dalsie request.
Requesty sa pridavaju cez metodu `Add` ci uz "zvonku` alebo z funkcie `processor`.

Cele to bezi paralelne a asynchronne.